woocommerce/woocommerce
# Refactor Revert button assets

### Changes proposed in this Pull Request:

This PR refactors the _Revert button_ assets used in WooCommerce templates so:

1. We don't need to register the script as having side effects.
2. We don't need to manually enqueue the script.
3. We can simplify `webpack-entries.js` a bit as well.
4. The stylesheet changes from `style.scss` to `editor.scss`, as it's only used in the editor.

Related: https://github.com/woocommerce/woocommerce/issues/42107.

### How to test the changes in this Pull Request:

1. Go to Appearance > Editor > Templates and select one of the WooCommerce templates, ie: Product Search Results.
2. If you have the 'blockfied' template, verify you have _Revert to Classic Product Template_ button.
![image](https://github.com/woocommerce/woocommerce/assets/3616980/7ed04bef-3805-4b2a-a729-799794db73bf)
3. If you don't have the 'blockfied' template, verify you have a button to transform the template into blocks:
![image](https://github.com/woocommerce/woocommerce/assets/3616980/831bc63d-7d57-4b5b-810e-6a58f4c70666)
6. Verify you can toggle between the two templates, the blockified one and the classic one.
7. Repeat the tests with all other WooCommerce templates: Product Catalog, Single Product, Products by Category, Products by Tag and Products by Attribute.
8. Go to any other template, ie: _Blog Home_ and verify the _Revert to Classic Product Template_ button is not there.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
